src/wp-includes/blocks.php

```diff
@@ -1502,12 +1502,61 @@ function parse_blocks( $content ) {
  */
 function do_blocks( $content ) {
 	$blocks = parse_blocks( $content );
+
+	/**
+	 * Filters the tree of raw parsed blocks before they are rendered.
+	 *
+	 * The parsed blocks form a hierarchy, rather than a flat array, as blocks can
+	 * be nested within each other. Also, take into account that calls to
+	 * `do_blocks` could be nested if there are blocks that use `do_blocks` in
+	 * their own rendering.
+	 *
+	 * Example:
+	 *
+	 *     add_filter( 'do_blocks_pre_render', function ( $blocks ) {
```

Comment:
It's not urgent given the discussion in Slack this-morning, but prior to 6.5 are you able to run this through the docs parser used by developer.wordpress.org to make sure the code example renders as expected.

There are instructions for setting up devhub at https://make.wordpress.org/docs/handbook/devhub/ -- it's been a while so I am not sure how up-to-date they are.

The WP-CLI command takes ages to run for the full code base, so  I suggest limiting it to this file only

Reply:
> it's been a while so I am not sure how up-to-date they are.

I spent some time trying to get it working and had a lot of trouble, finally gave up.

initially it broke with `yarn@4+`, had trouble with version mismatches between `wp-cli` and `composer` and their configs in the repo, had similar trouble building the developer theme, and its parent theme, had issues with PHP8 (understandable), and then once I got it building things, it seemed like it created a separate page for every docblock but I couldn't find any linked and organized pages as are on the production page.

if someone with experience and knowledge could review the README I'd be super grateful because I've really wanted to be able to preview my docs changes, and I don't like having to wait for releases to go make changes.

Reply:
> if someone with experience and knowledge could review the README I'd be super grateful because I've really wanted to be able to preview my docs changes, and I don't like having to wait for releases to go make changes.

I'll see if I can figure out the steps I took. It has been a while since I've needed to test docblocks.